app/controllers/mixins/actions/host_actions/discover.rb

```diff
@@ -25,46 +25,28 @@ def discover
-            if request.parameters[:controller] != "ems_cloud"
-              from_ip = params[:from_first].to_s + "." + params[:from_second].to_s + "." + params[:from_third].to_s + "." + params[:from_fourth]
-              to_ip   = params[:from_first].to_s + "." + params[:from_second].to_s + "." + params[:from_third].to_s + "." + params[:to_fourth]
+            from_ip = params[:from_first].to_s + "." + params[:from_second].to_s + "." + params[:from_third].to_s + "." + params[:from_fourth]
+            to_ip   = params[:from_first].to_s + "." + params[:from_second].to_s + "." + params[:from_third].to_s + "." + params[:to_fourth]
 
-              i = 0
-              while i < @discover_type.length
-                if @discover_type.length == 1 || params["discover_type_#{@discover_type[i]}"]
-                  discover_type.push(@discover_type[i].to_sym)
-                  @discover_type_checked.push(@discover_type[i])
-                end
-                i += 1
+            i = 0
+            while i < @discover_type.length
+              if @discover_type.length == 1 || params["discover_type_#{@discover_type[i]}"]
+                discover_type.push(@discover_type[i].to_sym)
+                @discover_type_checked.push(@discover_type[i])
               end
-
-              @from = {:first => params[:from_first], :second => params[:from_second], :third => params[:from_third], :fourth => params[:from_fourth]}
-              @to   = {:first => params[:from_first], :second => params[:from_second], :third => params[:from_third], :fourth => params[:to_fourth]}
+              i += 1
             end
+
+            @from = {:first => params[:from_first], :second => params[:from_second], :third => params[:from_third], :fourth => params[:from_fourth]}
+            @to   = {:first => params[:from_first], :second => params[:from_second], :third => params[:from_third], :fourth => params[:to_fourth]}
             @in_a_form = true
             drop_breadcrumb(:name => _("%{title} Discovery") % {:title => title}, :url => "/host/discover")
             @discover_type_selected = params[:discover_type_selected]
 
-            if request.parameters[:controller] == "ems_cloud" && params[:discover_type_selected] == 'azure'
-              @client_id = params[:client_id] if params[:client_id]
-              @client_key = params[:client_key] if params[:client_key]
-              @azure_tenant_id = params[:azure_tenant_id] if params[:azure_tenant_id]
-              @subscription = params[:subscription] if params[:subscription]
-
-              if @client_id == "" || @client_key == "" || @azure_tenant_id == "" || @subscription == ""
-                add_flash(_("Client ID, Client Key, Azure Tenant ID and Subscription ID are required"), :error)
-                render :action => 'discover'
-                return
-              end
-            elsif request.parameters[:controller] == "ems_cloud" || params[:discover_type_ipmi].to_s == "1"
```

Comment:
this code should have gone away too, right?

Before, we checked that ems_cloud is true, now we don't .. That doesn't seem to match all the rest of the PR where you're deleting stuff that was only for ems_cloud.